core/client/fs/src/main/java/alluxio/client/file/cache/PageStore.java

```diff
@@ -73,6 +94,104 @@ static PageStore create(AlluxioConfiguration conf) {
     return create(options);
   }
 
+  /**
+   * Checks if the data at the store location is compatible with the current configuration.
+   *
+   * @param conf the Alluxio configuration
+   * @return true if the data is compatible with the configuration, false otherwise
+   */
+  static boolean isCompatible(AlluxioConfiguration conf) {
+    String rootPath = conf.get(PropertyKey.USER_CLIENT_CACHE_DIR);
+    Path confPath = Paths.get(rootPath, CONF_FILE);
+    if (!Files.exists(confPath)) {
+      return false;
+    }
+    Properties properties = ConfigurationUtils.loadPropertiesFromFile(confPath.toString());
+    if (properties == null) {
+      return false;
+    }
+    AlluxioProperties alluxioProperties = new AlluxioProperties();
+    alluxioProperties.merge(properties, Source.DEFAULT);
+    AlluxioConfiguration cacheConf = new InstancedConfiguration(alluxioProperties);
```

Comment:
why we need to reload the cache conf from path? Can't we trust the argument passed in?

Reply:
Because configuration might changed between client restart and the data might not be compatible after the change. Here we just check the stored configuration and discard the data if the configuration is not compatible(e.g. page size changed, store type changed).